processing/data_collection/gazette/spiders/sp_santo_andre.py

```diff
@@ -0,0 +1,187 @@
+import dateparser
+from datetime import datetime, date
+from urllib.parse import unquote
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+from scrapy import FormRequest, Request
+from scrapy.selector.unified import Selector
+from gazette.settings import FILES_STORE
+import re
+from w3lib.html import remove_tags
+from pathlib import Path
+from typing import Optional
+from hashlib import sha1
+
+
+class SpSantoAndreSpider(BaseGazetteSpider):
+    name = "sp_santo_andre"
+    TERRITORY_ID = "3547809"
+    custom_settings = {
+        "ITEM_PIPELINES": {
+            "gazette.pipelines.GazetteDateFilteringPipeline": 50,
+            "gazette.pipelines.ExtractTextPipeline": 200,
+        }
+    }
+
+    allowed_domains = ["santoandre.sp.gov.br"]
+    start_urls = [
+        "http://www.santoandre.sp.gov.br/publicacao/edicao/consultaedicao.aspx"
+    ]
+    start_date = date(2010, 1, 1)
+    TODAY = datetime.now().strftime("%d/%m/%Y")
+    JAVASCRIPT_POSTBACK_REGEX = r"javascript:__doPostBack\('(.*)',''\)"
+    CURRENT_PAGE = 1
+
+    def start_requests(self):
+        for url in self.start_urls:
+            yield Request(url, callback=self.search_by_date)
+
+    def _get_form_params(self, response):
+        view_state = response.css("#__VIEWSTATE::attr(value)").get()
+        event_validation = response.css("#__EVENTVALIDATION::attr(value)").get()
+        return view_state, event_validation
+
+    def search_by_date(self, response):
+        VIEW_STATE, EVENT_VALIDATION = self._get_form_params(response)
+        yield FormRequest.from_response(
+            response,
+            callback=self.parse,
+            formname="aspnetForm",
+            formdata={
+                "__EVENTTARGET": "",
+                "__EVENTARGUMENT": "",
+                "__VIEWSTATE": VIEW_STATE,
+                "__EVENTVALIDATION": EVENT_VALIDATION,
```

Comment:
Since you are working with `FormRequest.from_response` and those `"__SOMETHING"` keys are "hidden input elements" you normally wouldn't need to manually capture its values, `from_response` already should do that for you :)

I tested it here and it worked. Could you check this too?

On the other requests, the `"__EVENTTARGET"` key was needed, though.

Reply:
> Since you are working with `FormRequest.from_response` and those `"__SOMETHING"` keys are "hidden input elements" you normally wouldn't need to manually capture its values, `from_response` already should do that for you :)
> 
> I tested it here and it worked. Could you check this too?
> 
> On the other requests, the `"__EVENTTARGET"` key was needed, though.

I had not aware of the `FormRequest.from_response `capabilities. I already tested your suggestion and worked fine!